### ISSUE TITLE
Field error returned from Custom Script Adapter is ignored

### ISSUE THREAD

**Issue by jessesnyder** [collective/collective.easyform#217]
## Steps to reproduce
1. Add a custom script adapter with the following contents, on a default form with a `subject` field: 
```
return {'topic': 'An error message!'}
```

2. Submit the completed form

## Expected behavior
The "Subject" field should display an error

## Actual behavior
The form submits successfully with no error displayed.

**Comment by jessesnyder** [comment]
Checking to see if adapters have already set field validation errors before re-running form validation seems to remedy the problem (edit to `EasyFormForm.edit()`):
````
    def update(self):
        """ Update form - see interfaces.IForm """
        self.formMaybeForceSSL()
        super(EasyFormForm, self).update()
        self.template = self.form_template
        if self.request.method != "POST" or self.context.thanksPageOverride:
            # go with all but default thank you page rendering
            return
        # If an adapter has already set errors, don't re-run extraction and
        # validation, just bail out:
        if self.widgets.errors:
            return
        data, errors = self.extractData()
        if errors:
            # render errors
            return
```

**Comment by wobsta** [comment]
fixed by #238
